<launch>
  
  <arg name="gui" value="true" />
  <!-- <arg name="input_cloud" value="/kinect_head/depth_registered/points" /> -->
  <!-- <arg name="input_cloud" value="/kinect_head/depth_registered/quater/points" /> -->
  <arg name="input_cloud" value="/kinect_head/depth_registered/points" />
  <arg name="manager_name" default="hand_navigation_pr2_manager" />
  <arg name="manager" default="/$(arg manager_name)" />

  <node name="$(arg manager_name)"
        pkg="nodelet" type="nodelet"
        args="manager"/>

  <arg name="input_camera_info" value="/kinect_head/rgb/camera_info" />
  <arg name="input_rgb_image" value="/kinect_head/rgb/image_rect_color" />
  <arg name="input_depth_image" value="/kinect_head/depth_registered/hw_registered/image_rect" />
  <arg name="input_pose_array" value="/edgetpu_human_pose_estimator/output/poses" />


  <node name="people_pose_2d_to_3d"
        pkg="jsk_perception" type="people_pose_2d_to_3d.py"
        output="screen">
    <remap from="~input/pose" to="$(arg input_pose_array)" />
    <remap from="~input/info" to="$(arg input_camera_info)" />
    <remap from="~input/depth" to="$(arg input_depth_image)" />
    <rosparam subst_value="true">
      approximate_sync: true
      queue_size: 100
    </rosparam>
  </node>

  <node name="people_poses_to_poses"
        pkg="jsk_recognition_msgs" type="people_pose_array_to_pose_array.py">
    <remap from="~input" to="people_pose_2d_to_3d/output/pose" />
  </node>

  <node name="arm_pose_to_boxes"
        pkg="hand_navigation_pr2" type="arm_pose_to_boxes.py">
    <remap from="~input" to="people_pose_2d_to_3d/output/pose" />
  </node>

  <node name="attention_clipper"
        pkg="nodelet" type="nodelet"
        args="load jsk_pcl/AttentionClipper $(arg manager)"
        clear_params="true" >
    <remap from="~input/points" to="$(arg input_cloud)" />
    <remap from="~input/box_array" to="/arm_pose_to_boxes/output/boxes" />
    <rosparam>
      use_multiple_attention: true
      approximate_sync: true
      queue_size: 1000
    </rosparam>
  </node>

  <node name="extract_indices"
        pkg="nodelet" type="nodelet"
        args="load jsk_pcl/ExtractIndices $(arg manager)"
        clear_params="true">
    <remap from="~input" to="$(arg input_cloud)" />
    <remap from="~indices" to="attention_clipper/output/point_indices" />
    <remap from="~output" to="/attention_cloud" />
    <rosparam>
      keep_organized: true
      approximate_sync: true
      queue_size: 100
    </rosparam>
  </node>

  <node name="cluster_point_indices_decomposer"
        pkg="nodelet" type="nodelet"
        args="standalone jsk_pcl/ClusterPointIndicesDecomposer">
    <remap from="~input" to="$(arg input_cloud)" />
    <remap from="~target" to="attention_clipper/output/cluster_point_indices" />
     <rosparam subst_value="true">
      <!-- align_boxes: true -->
      <!-- align_boxes_with_plane: false -->
      <!-- use_pca: $(arg use_pca) -->
      <!-- target_frame_id: head_mount_kinect_rgb_optical_frame -->
      approximate_sync: true
    </rosparam>
  </node>
  
 <!-- <node name="attention_estimation"  -->
 <!--       pkg="hand_navigation_pr2" type="attention_estimation.py"> -->
 <!-- </node> -->
  
  <group if="$(arg gui)" >
    <node name="$(anon rviz)"
          pkg="rviz" type="rviz"
          args="-d $(find hand_navigation_pr2)/launch/config/arm_pose.rviz" >
    </node>
  </group>
</launch>
